android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -184,14 +195,70 @@ protected void dispatchOnDisappear() {
         .getEventDispatcher()
         .dispatchEvent(new ScreenDisappearEvent(mScreenView.getId()));
 
+    dispatchEventInChildContainers(ScreenLifecycleEvent.Disappear);
+
+    dispatchTransitionProgress(1.0f, true, true);
+  }
+
+  public void incrementCoalescingKey() {
+    mCoalescingKey++;
+  }
+
+  protected void dispatchTransitionProgress(float alpha, boolean closing, boolean shouldIncrementCoalescingKey) {
+    if (shouldIncrementCoalescingKey) {
+      incrementCoalescingKey();
+    }
+    sendTransitionProgressEvent(alpha, closing);
+
+    if (mAboveScreen != null && mAboveScreen.getFragment() != null && !mAboveScreen.getFragment().isSendingProgress()) {
```

Comment:
Can you add a comment explaining this here. Why would more than one view send progress at a given time? Is there a problem when they do? I understand that progress updates will be delivered to separate views

Reply:
We send the transition progress to both of the transitioning screens. We do it by subscribing to the ongoing transition of fragments: https://github.com/software-mansion/react-native-screens/pull/890/files#diff-2c98f4cb51ad28895af60cd79d14b67a66256480a554d81a9dfa6f51c84061c2R41. But if we are pushing a `transparentModal`, the fragment below does not participate in the transition, so it does not call the `onAnimationStart` method. In order to achieve the state when always 2 views send progress during the transition, we check if above screen is sending progress, and, if not, we do it from the screen being pushed. Same applies to modals on iOS.